cips/cip-18.md

```diff
@@ -38,49 +38,62 @@ The following API is proposed for the standardised gas estimation service.
 
 ```proto
 service GasEstimator {
-    rpc EstimateGasUsage(EstimateGasUsageRequest) returns (EstimateGasUsageResponse) {}
-    rpc EstimateGasPrice(EstimateGasPriceRequest) returns (EstimateGasPriceResponse) {}
+    rpc EstimateFee(EstimateFeeRequest) returns (EstimateFeeResponse) {}
+    rpc EstimateFeeAndGas(EstimateFeeAndGasRequest) returns (EstimateFeeAndGasResponse) {}
 }
 
-message EstimateGasUsageRequest {
-    cosmos.tx.Tx tx = 1;
+enum TxPriority {
+  NONE = 0;
+  LOW = 1;
+  MEDIUM = 2;
+  HIGH = 3;
 }
 
-message EstimateGasUsageResponse {
-    uint64 estimated_gas_used = 1;
+message EstimateFeeRequest {
+    TxPriority tx_priority = 1;
 }
 
-message EstimateGasPriceRequest {}
+message EstimateFeeResponse {
+    double estimated_gas_price = 1;
+}
+
+message EstimateFeeAndGasRequest {
+    TxPriority tx_priority = 1;
+    cosmos.tx.Tx tx = 2;
+}
 
-message EstimateGasPriceResponse {
+message EstimateFeeAndGasResponse {
     double estimated_gas_price = 1;
+    uint64 estimated_gas_used = 2;
 }
+
+
 ```
 
 Given it's wide usage both in the Cosmos SDK and more broadly, the service would be implemented using gRPC. RESTful endpoints may optionally be supported in the future using something like `grpc-gateway`.
 
 Given the expected reliance on clients, all future changes must be done in a strictly non-breaking way.
 
+The notion of urgency, represented by the `TxPriority` field is both simple and subjective. Loosely it means:
+
+- High: whithin 1 block
+- Medium: within a few blocks
+- Low: withing the next hour
```

Comment:
Just to confirm what these mean from an implementation point of view (if this is the right place to discuss or we can leave this for an ADR):

- High: means returning the median gas of the highest 10% gas prices in the last 5 blocks, for example?
- Medium: would be the median of all gas prices in the last 5 blocks
- Low: would be the median of the lowest 10% gas prices in the last 5 blocks?

Reply:
This should be covered in the celestia-app ADR